multiqc/modules/verifybamid/verifybamid.py

```diff
@@ -0,0 +1,238 @@
+        #create a dictionary to populate from this sample's file
+        parsed_data = dict()
+        # set a empty variable which denotes if the headers have been read
+        headers = None
+        # for each line in the file
+        for l in f['f'].splitlines():
+                # split the line on tab
+                s = l.split("\t")
+                # if we haven't already read the header line
+                if headers is None:
+                    # remove the leading hash from the first element of the header
+                    s[0] = s[0].lstrip('#')
+                    # assign this list to headers variable
+                    headers = s
+                # for all rows after the first
+                else:
+                    s_name=s[0]
+                    s_name = self.clean_s_name(s_name, f['root'])
+                    # create a dictionary entry with the first column as a key (sample name) and an empty dictionary as a value
+                    parsed_data[s_name] = {}
+                    # for each item in list of items in the row
+                    for i, v in enumerate(s):
+                    # if it's not the first element (if it's not the name)
+                        if i != 0:
+                            # try and convert the value into a float
+                            try:
+                            # and add to the dictionary the key as the corrsponding item from the header and the value from the list
+                                parsed_data[s_name][headers[i]] = float(v)
+                                #if can't convert to float...
+                            except ValueError:
+                            # add to the dictionary the key as the corrsponding item from the header and the value from the list
+                                parsed_data[s_name][headers[i]] = v
+    
+        # nothing has been parsed return None
+        if len(parsed_data) == 0:
+            return None
+        # else return the dictionary
+        return parsed_data
+
+    def verifybamid_general_stats_table(self):
+        """ Take the percentage of contamination from all the parsed *.SELFSM files and add it to the basic stats table at the top of the report """
+
+        # create a dictionary to hold the columns to add to the general stats table
+        headers = OrderedDict()
+        # available columns are:
+        #SEQ_ID RG  CHIP_ID #SNPS   #READS  AVG_DP  FREEMIX FREELK1 FREELK0 FREE_RH FREE_RA CHIPMIX CHIPLK1 CHIPLK0 CHIP_RH CHIP_RA DPREF   RDPHET  RDPALT
+        #,see https://genome.sph.umich.edu/wiki/VerifyBamID#Interpreting_output_files
+
+        # add the CHIPMIX column. set the title and description
+        headers['CHIPMIX'] = {
+            'title': 'Contamination Prediction - sequence and array',
+            'description': 'VerifyBamID: CHIPMIX -   Sequence+array estimate of contamination (NA if the external genotype is unavailable) (0-1 scale)',
```

Comment:
It would be nice to add a check to see whether all samples are `NA` and to hide the column if so. eg:

```python
'hidden': all( [ s['CHIPMIX'] == 'NA' for s in self.verifybamid_data.values() ] )
```

One-liner off the top of my head without testing, so may not work. But basically set `hidden` to `True` in the config if everything is `NA`.

Reply:
Coming back to this after going through the rest of the code - it may be worth defining an extra variable per sample in the parsing so that we have `self.verifybamid_data[s_name]['has_chip_data']` or something (as a boolean). Then it's easier to refer to repeatedly in this kind of context.

Reply:
Also, if it's _really_ not interesting (which I guess is the case if it's all just `NA`?)  then we could skip having this in the plot at all, even hidden. So only add the column if we have some data.